src/docfx/resolve/Resolve.cs

```diff
@@ -2,82 +2,88 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.IO;
 
 namespace Microsoft.Docs.Build
 {
-    internal delegate (string content, Document path) ResolveContent(Document relativeTo, string href);
-
-    internal delegate string ResolveHref(Document relativeTo, string href, Document resultRelativeTo);
-
     internal static class Resolve
     {
         public static (string content, Document file) TryResolveContent(this Document relativeTo, string href)
         {
-            var buildItem = TryResolveHref(relativeTo, href);
-            if (buildItem == null)
-            {
-                return default;
-            }
+            var (file, _) = TryResolveFile(relativeTo, href);
 
-            return (buildItem.ReadText(), buildItem);
+            return file != null ? (file.ReadText(), file) : default;
         }
 
         public static (string href, Document file) TryResolveHref(this Document relativeTo, string href, Document resultRelativeTo = null)
         {
-            var file = TryResolveHref(relativeTo, href);
+            var (file, fragmentQuery) = TryResolveFile(relativeTo, href);
+
+            // Cannot resolve the file, leave href as is
             if (file == null)
             {
                 return (href, null);
             }
 
-            var (_, fragment, query) = HrefUtility.SplitHref(href);
-
-            var resolvedHref = Uri.EscapeUriString(file.SiteUrl) + fragment + query;
-            if (href[0] == '/')
-            {
-                return (resolvedHref, file);
-            }
+            var resolvedHref = file.SiteUrl + fragmentQuery;
 
             if (resultRelativeTo == null)
             {
                 return (resolvedHref, file);
             }
 
+            // Make result relative to `resultRelativeTo`
             resolvedHref = PathUtility.GetRelativePathToFile(resultRelativeTo.SiteUrl, file.SiteUrl).Replace('\\', '/');
 
-            return (Uri.EscapeUriString(resolvedHref) + fragment + query, file);
+            return (resolvedHref + fragmentQuery, file);
         }
 
-        private static Document TryResolveHref(this Document relativeTo, string href)
+        private static (Document file, string fragmentQuery) TryResolveFile(this Document relativeTo, string href)
         {
             if (string.IsNullOrEmpty(href))
             {
                 return default;
             }
 
-            if (!HrefUtility.IsRelativeHref(href))
+            var (path, fragment, query) = HrefUtility.SplitHref(href);
```

Comment:
>path [](start = 17, length = 4)

need url decode href to get path?

Reply:
Right, I left a TODO in Document to handle URL encoding/decoding. They will be handled later.

---
In reply to: [193278818](https://github.com/dotnet/docfx/pull/2839#discussion_r193278818) [](ancestors = 193278818)